src/lib-dcrypt/dcrypt-openssl1.c

```diff
@@ -3639,7 +3639,12 @@ dcrypt_openssl_public_key_id_old(struct dcrypt_public_key *key,
 	/* digest this */
 	SHA256((const unsigned char*)pub_pt_hex, strlen(pub_pt_hex), buf);
 	buffer_append(result, buf, SHA256_DIGEST_LENGTH);
-	OPENSSL_free(pub_pt_hex);
+
+	/* XD curve uses binary_to_hex_ucase, no OPENSSL_free */
+#ifdef HAVE_X25519
+	if (!IS_XD_CURVE(nid))
+#endif
+		OPENSSL_free(pub_pt_hex);
```

Comment:
Isn't this going to complain because pub_pt_hex is const pointer, or does OPENSSL_free() just cast it away? Anyway, perhaps nicer:

```
	const char *pub_pt_hex = NULL;
	const char *pub_pt_hex_free = NULL;
...
		pub_pt_hex = pub_pt_hex_free =
			ec_key_get_pub_point_hex(EVP_PKEY_get0_EC_KEY(pub));
...
OPENSSL_free(pub_pt_hex_free);
```

Would also avoid the secondary ifdef/if checks.